src/v/raft/consensus.cc

```diff
@@ -510,7 +510,7 @@ ss::future<result<model::offset>> consensus::linearizable_barrier() {
         model::term_id term;
     };
 
-    std::optional<ss::semaphore_units<>> u = co_await _op_lock.get_units();
+    ss::semaphore_units<> u = co_await _op_lock.get_units();
```

Comment:
> r/consensus: do not wrap units in optional

why, cleanup? or was it unnecessary or for some other reason?

Reply:
I used this to release units, however units have `return_all()` method witch returns all the units and signal semaphore.

Reply:
cool